tests/lean/run/toFromJson.lean

```diff
@@ -0,0 +1,83 @@
+import Lean
+
+open Lean
+open Lean Parser Term
+
+declare_syntax_cat json
+syntax strLit : json
+syntax numLit : json
+syntax "{" (Lean.Parser.ident ": " json),* "}" : json
+syntax "[" json,* "]" : json
+
+syntax "json! " json : term
```

Comment:
We don't use `!` as a macro suffix anymore, but I suppose it's just a test

Reply:
I was thinking also `j#` a bit like a string literal, this would be json literal, but it looked really awkward because you don't have quoted. If I'm going to touch the code again, what do you suggest?

Reply:
I've made `json` for now.

Reply:
It really doesn't matter here, but what do you think about `` `[Json|...] ``? It makes more sense for DSLs that are not already properly parenthesized, but it's my favorite style so far.